packages/editor/src/managers/SceneManager.ts

```diff
@@ -186,24 +193,38 @@ export class SceneManager {
    * @return {Promise}        [generated screenshot according to height and width]
    */
   async takeScreenshot(width: number, height: number) {
-    const { screenshotRenderer } = this
-    const sceneNode = Engine.scene as any as SceneNode
     const originalRenderer = Engine.renderer
-    Engine.renderer = screenshotRenderer
+    Engine.renderer = this.screenshotRenderer
     this.disableUpdate = true
-    let scenePreviewCamera = sceneNode.findNodeByType(ScenePreviewCameraNode)
-    if (!scenePreviewCamera) {
-      scenePreviewCamera = new ScenePreviewCameraNode()
-      Engine.camera.matrix.decompose(scenePreviewCamera.position, scenePreviewCamera.rotation, scenePreviewCamera.scale)
-      CommandManager.instance.executeCommandWithHistory(EditorCommands.ADD_OBJECTS, [scenePreviewCamera])
+
+    const entities = getAllEntitiesWithComponent(ScenePreviewCameraTagComponent)
```

Comment:
We have Engine.activeCameraEntity for storing the location camera entity, we should leverage this in the editor too for this